mllib/src/main/scala/org/apache/spark/ml/feature/QuantileDiscretizer.scala

```diff
@@ -49,6 +49,21 @@ private[feature] trait QuantileDiscretizerBase extends Params
 
   /** @group getParam */
   def getNumBuckets: Int = getOrDefault(numBuckets)
+
+  /**
+   * Relative error (see approxQuantile documentation for description).  Must be >= 0.
+   * By default, the relative error will be in the range [0.001, 0.01]. The exact value
+   * is computed based on the number of buckets, decreasing as the number of buckets
+   * increases. Explicitly setting the relative error will override this default calculation.
+   * @group param
+   */
+  val relativeError = new DoubleParam(this, "relativeError", "The relative target precision " +
+    "for approxQuantile",
+    ParamValidators.inRange(0.0, 1.0))
+
+  /** @group Param */
+  def getRelativeError: Double = get(relativeError).getOrElse
```

Comment:
Please return `$(relativeError)` directly because there are other places to fetch the param value.